shared/src/commonMain/kotlin/io/customer/shared/di/DIGraph.kt

```diff
@@ -0,0 +1,101 @@
+package io.customer.shared.di
+
+import kotlin.reflect.KClass
+
+/**
+ * Base dependency injection class to satisfy dependencies without adding any external dependency.
+ * Any DI component created within the SDK should inherit this class.
+ *
+ * @see [getNewInstance] and [getSingletonInstance] to get more details on how to use this class.
+ */
+abstract class DIGraph {
+    /**
+     * Map of overridden dependencies, helpful in mocking for test functions.
+     */
+    val overrides: MutableMap<String, Any> = mutableMapOf()
+
+    /**
+     * Call when you want to override a dependency with a mock in test functions.
+     * `di.overrideDependency(CustomerIOInstance::class.java, customerIOMock)`
+     *
+     * See `CustomerIOTest` (or any other class that is top-level of the module) for an example on use.
+     */
+    fun <DEP : Any> overrideDependency(dependency: KClass<DEP>, value: DEP) {
+        overrides[dependency.getClassNameOrThrow(value = value)] = value as Any
```

Comment:
```suggestion
        overrides[dependency.getClassNameOrThrow(value = value).simpleName] = value as Any
```

I think you need this change so the `inline fun override()` works as it uses `simpleName`? Maybe you don't need if `getClassNameOrThrow()` returns a simple name.

Reply:
`simpleName` is already there and returns nullable names. Please read documentation on `getClassNameOrThrow` to see why I added this method. But yes, we can give it better names. How does `className` sounds?